include/cadmium/core/modeling/coupled.hpp

```diff
@@ -270,6 +272,126 @@ namespace cadmium {
 			}
 			addCoupling(EOC, portFrom, portTo);
 		}
+
+
+		void flatten(){
+        	std::vector<std::shared_ptr<Coupled>> toFlatten;
+        	for (auto& component: components) {
+        		auto coupled = std::dynamic_pointer_cast<Coupled>(component);
+        		if (coupled != nullptr) {
+        			toFlatten.push_back(coupled);
+        		}
+        	}
+
+        	for(auto itc = toFlatten.begin(); itc != toFlatten.end(); itc++){
```

Comment:
why don't you use `for (auto& itc: toFlatten)`?

Reply:
I modified it when possible. When we need to eliminate elements is easier to have iterators to use erase(position).